api/helpers/uploadedFileErrorChecks.js

```diff
@@ -0,0 +1,270 @@
+const NodeClam = require('clamscan');
+const { resolve } = require('path');
+const AWS = require('aws-sdk');
+const s3 = new AWS.S3({ region: 'eu-west-2' });
+
+const deleteFileFromStorage = require('./deleteFileFromStorage');
+
+const TWO_HUNDRED_MEGABYTES = 200 * 1_000_000;
+const MAX_BYTES_PER_FILE = TWO_HUNDRED_MEGABYTES;
+
+const inDevEnvironment = process.env.NODE_ENV === 'development';
+let clamscan;
+
+async function connectToClamAV(req) {
+    try {
+        const { clamav_enabled: clamavEnabled } = req._sails.config.eAppS3Vals;
+        const clamavEnabledStringToBool = JSON.parse(clamavEnabled);
+
+        if (!clamavEnabledStringToBool) {
+            throw new Error('ClamAV is not enabled.');
+        }
+
+        sails.log.info('Connecting to clamAV...');
+        clamscan = await initialiseClamScan(req);
+        sails.log.info('Connected successfully 🎉');
+
+        return true;
+    } catch (err) {
+        sails.log.error(`Clamav connection unavailable. ${err}`);
+        return false;
+    }
+}
+
+function initialiseClamScan(req) {
+    const { clamav_host: clamavHost, clamav_port: clamavPort } =
+        req._sails.config.eAppS3Vals;
+
+    const clamAvOptions = {
+        clamdscan: {
+            host: inDevEnvironment ? '127.0.0.1' : clamavHost,
```

Comment:
I think this part is unnecessary. If in dev mode, you aren't guaranteed to run on 127.0.0.1. I have my clamav running on a different docker network and need to comment this bit out to force it to use clamavHost.

Using the value from the env file should be fine in all cases.

Reply:
Very true, I honestly don't know why I did it like this 🤦